src/main/c/functions.c

```diff
@@ -1,44 +1,159 @@
 #include <math.h>
 #include <stdlib.h>
+#include <string.h> // For memset.
 #include "data.h"
 #include "functions.h"
 
+
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+#endif
+
 /**
  * Returns the size of a triangular matrix, without the diagonal.
  */
-size_t size_triangular_matrix(const size_t n) {
+size_t
+size_triangular_matrix(const size_t n) {
   return ((n - 1) * n) / 2;
 }
 
+/**
+ * Computes the distance between two particles.
+ */
+double
+compute_distance(const Particle *p1, const Particle *p2) {
+  const double x_diff = p1->x_coordinate - p2->x_coordinate;
```

Comment:
It is not clear for me why to use the keyword _const_ in the declaration of the variables _x\_diff_ and _y\_diff_. They are local variables and they are not used anywhere else in the code. So I think that it would be better to just leave them as _double_s.

Reply:
First, those are not **"C"** standards, are gnu standards _(two completely different things)_. For example, the linux kernel does not follow such standard. And if we are going to follow such standard, we should follow it correctly, meaning the braces should start in a new line and apply the changes everywhere.
Anyways, those changes should be discussed by the team, the only reason standards exist is to ensure consistency across the code base, making it easier to read. As such, you should had done that on a different branch, not on a existing one _(which does not have nothing to do with this)_.

Thus, I believe the best thing is to revert those commits and create a new PR to propose the new standard. On that PR apply the changes everywhere.

-----

Second, answering your question, because they are immutable, they are meant to be computed and never modified again. Making them `const` will ensure that, if by accident anyone modify them, the compiler will warn.
Minimizing the mutable state of each function, leads to more clear code which is easier to maintain.